src/rt/cpp/when.h

```diff
@@ -14,13 +14,102 @@ namespace verona::cpp
 {
   using namespace verona::rt;
 
+  template<typename T>
+  struct cown_ptr_span
+  {
+    cown_ptr<T>* array;

```

Comment:
Could you add comments about ownership here? Is this class opening the underlying cownptrs or borrowing them. Explaining that, and then adding comments elsewhere. 

Does std::move make sense if they are borrowed? Hadn't thought about this for my earlier review.

Reply:
I think it's clearer if `cown_ptr_span` always owns the included cowns. This can happen either by allocating a new cown_ptr array and incrementing the reference count, or by using the template parameter and passing an array of cown_ptr that already holds the ownership.

I added a comment that explains that.